### PR TITLE
bug 1801397: wait for membership before starting

### PR DESCRIPTION
**Pull request by deads2k**
make the static pod that we want to use runnable.

### CODE REVIEW

File: bindata/etcd/pod.yaml [hunk 1]
```diff
@@ -20,16 +20,56 @@ spec:
...
+        initial_cluster=""
+        member_output=$( ${ETCDCTL} member list | cut -d',' -f3 )
+        for endpoint_key in ${member_output}; do
+          endpoint=$(${ETCDCTL} member list | grep $endpoint_key | awk -F'[, ]' '{ print $7 }')
+          initial_cluster+="$endpoint_key=$endpoint,"
+          echo "adding $endpoint_key=$endpoint,"
+        done
+        # trim last comma
+        initial_cluster="${initial_cluster::-1}"
+        echo $initial_cluster
```
Comment (retroflexer) [inline bindata/etcd/pod.yaml]: Not sure why the `for` loop is needed.

File: bindata/etcd/pod.yaml [hunk 2]
```diff
@@ -20,16 +20,56 @@ spec:
...
+        ETCDCTL="etcdctl --cacert=/etc/kubernetes/static-pod-resources/configmaps/etcd-serving-ca/ca-bundle.crt \
+                           --cert=/etc/kubernetes/static-pod-resources/secrets/etcd-all-peer/etcd-peer-NODE_NAME.crt \
+                           --key=/etc/kubernetes/static-pod-resources/secrets/etcd-all-peer/etcd-peer-NODE_NAME.key \
+                           --endpoints=${ALL_ETCD_ENDPOINTS}"
+        ${ETCDCTL} member list
+
+        echo "waiting for member NODE_NAME..."
+        COUNT=30
+        while [ $COUNT -gt 0 ]; do
+          IS_MEMBER_PRESENT=$(${ETCDCTL} member list | grep -o "NODE_NAME.*:2380")
```
[the remark this answers is not in the document]
Comment (hexfusion) [reply bindata/etcd/pod.yaml]: Also, I believe the while loop will exit 1 when it completes and not go any further.
```
 done || {
           echo "member NODE_NAME is not present after 30 seconds"
           exit 1
   }
```

**Comment by deads2k** [comment]
superceded

**Pull request closed without merging**